From 8334efd3defcdd8bf6a1c9cb497e8d606cc51e54 Mon Sep 17 00:00:00 2001
From: Romain Pokrzywka <romain.pokrzywka@kdab.com>
Date: Thu, 13 Sep 2012 11:17:23 -0700
Subject: [PATCH] Fix gstreamer plugin build for platforms with qreal -> float

simple case of qFuzzyCompare with double and float.
I kept m_volume as double rather than changing it to float, to mimic
the behavior of the mute variable.

Change-Id: Ife2f58f60738215ebd430ba94bef885c09994e72
---
 .../mediacapture/qgstreamercapturesession.cpp      |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/../qt5-src/qtmultimedia/src/plugins/gstreamer/mediacapture/qgstreamercapturesession.cpp b/../qt5-src/qtmultimedia/src/plugins/gstreamer/mediacapture/qgstreamercapturesession.cpp
index 435a413..cc3965e 100644
--- a/../qt5-src/qtmultimedia/src/plugins/gstreamer/mediacapture/qgstreamercapturesession.cpp
+++ b/../qt5-src/qtmultimedia/src/plugins/gstreamer/mediacapture/qgstreamercapturesession.cpp
@@ -1025,7 +1025,7 @@ void QGstreamerCaptureSession::setMuted(bool muted)
 
 void QGstreamerCaptureSession::setVolume(qreal volume)
 {
-    if (!qFuzzyCompare(volume, m_volume)) {
+    if (!qFuzzyCompare(double(volume), m_volume)) {
         m_volume = volume;
         if (m_audioVolume)
             g_object_set(G_OBJECT(m_audioVolume), "volume", m_volume, NULL);
-- 
1.7.9.5

